src/test/scala/com/fulcrumgenomics/util/MetricTest.scala

```diff
@@ -31,6 +31,8 @@ import com.fulcrumgenomics.testing.UnitSpec
 
 private case class TestMetric(foo: String, bar: Int, car: String = "default") extends Metric
 
+private case class TestMetricWithOption(foo: String, bar: Int, option: Option[String]) extends Metric
```

Comment:
Could you add an `Option[Int]` or other non-String type as well please?  Would be good to test that it works for things that need type converting too.